src/core/bittorrent/session.cpp

```diff
@@ -1443,8 +1438,20 @@ void Session::setAppendLabelToSavePath(bool append)
 {
     if (m_appendLabelToSavePath != append) {
         m_appendLabelToSavePath = append;
-        foreach (TorrentHandlePrivate *const torrent, m_torrents)
-            torrent->handleDefaultSavePathChanged();
+        foreach (TorrentHandle *const torrent, m_torrents) {
+            QString label = torrent->label();
+            if (label.isEmpty()) continue;
+
+            QString testedOldSavePath = m_defaultSavePath;
+            QString newSavePath = m_defaultSavePath;
```

Comment:
>Do I understand correctly the below lines? They remove the label from the path for ALL torrents if the feature is disabled and they add it for ALL torrents if enabled, right?
In case we didn't discuss this change in #3495 then that's ok.

Yes, it affects all torrents (with default save path). And yes, we discussed this in #3495. This way does not coincide with my vision (see #3495). It's just a return to the old behavior.